openstack/auth_env.go

```diff
@@ -22,6 +39,16 @@ func AuthOptionsFromEnv() (gophercloud.AuthOptions, error) {
 	domainID := os.Getenv("OS_DOMAIN_ID")
 	domainName := os.Getenv("OS_DOMAIN_NAME")
 
+	// If OS_PROJECT_ID is set, overwrite tenantID with the value.
+	if v := os.Getenv("OS_PROJECT_ID"); v != "" {
```

Comment:
Gophercloud uses `TenantName` and `TenantID` instead of `ProjectName` and `ProjectID`. Therefore, there's no way to have a conflict between `TenantName` and `ProjectName`.